[ENH][systoleos] First commit

oeramo opened the pull request:
Progress up to this point tracked in https://github.com/SystoleOS/guix-systole/pull/52

oeramo committed:
[ENH][systoleos] First commit

Progress up to this point tracked in https://github.com/SystoleOS/guix-systole/pull/52

oeramo commented:
@RafaelPalomar I have made a mistake with the other branch that is currently set as default, https://github.com/OUH-MESHLab/BrainlabMirror/tree/supercede-systoleos-issue. As the admin, please set `main` as the default branch and delete the `supercede-systoleos-issue` branch. Thank you.

RafaelPalomar commented on systoleos.scm:
```diff
@@ -0,0 +1,261 @@
+;; This operating-system image definition can be built with:
+;; `guix system image -L . -t iso9660 guix-systole/systoleos/systoleos.scm`
+
```
This probably needs to be updated to the new path

RafaelPalomar replied:
Also. How about the `guix vm` command

asiangoldfish replied:
There doesn't seem to be a `guix vm` command. My version of guix: `8ee456e2bda8f72ccaf2398a1709a85e6e32d952`.

RafaelPalomar reviewed (changes requested):
@oeramo, please add a `README.md` with some notes on how to build and run the image with `qemu`

oeramo committed:
[ENH][systoleos] Separate base os definition and full for bare metal

oeramo committed:
[ENH][systoleos] Include background artwork

oeramo committed:
[WIP][systoleos] Remove redundant imports¨

oeramo committed:
[WIP][systoleos] Sync progress, inheriting one os definition into the other does not work

Building with command:
```
time guix time-machine -C systoleos/channels.scm --substitute-urls='https://ci.guix.gnu.org https://bordeaux.guix.gnu.org https://substitutes.nonguix.org' -- system image -L . -t iso9660 systoleos/brainlabmirror.scm
```

Will revert to two independent definitions, remove inheritance.

oeramo committed:
[WIP][systoleos] Make the different os definitions independent

oeramo committed:
[WIP][systoleos] Use install script

oeramo commented:
@RafaelPalomar, I haven't managed to find a way to build the image from within this repository.

I had done my previous tests on the [guix-systole](https://github.com/SystoleOS/guix-systole/pull/48) repository before moving the PR here, and there building the image was trivial because all dependencies were under the same repository, and I could use the `-L` flag of `guix system image` to specify where to look for Scheme packages.

When the operating-system is defined in a different location, to avoid having to manually clone dependencies in place, we should leverage Guix Channels to find the dependencies. I have tried to set this up in a few ways, but the builder cannot find 3D Slicer, despite it being correctly imported, and despite the `guix-systole` channel pulling successfully before building starts:
```
guest@gnu ~/systole-playground/BrainlabMirror$ guix time-machine -C systoleos/channels.scm -- system image -t iso9660 systoleos/brainlabmirror.scm
ice-9/eval.scm:223:20: In procedure proc:
error: slicer-5.8: unbound variable
hint: Did you forget `(use-modules (guix-systole packages slicer))'?
```

Furthermore, when the `operating-system` definitions are set to depend on each other (one inherits the other, like in https://github.com/OUH-MESHLab/BrainlabMirror/pull/1/commits/23636a23d92d22a9137b3b42959b924c01bc7ba8), the builder cannot find the inherited module:
```
guest@gnu ~/systole-playground/BrainlabMirror$ time guix system image -L . -t iso9660 systoleos/brainlabmirror.scm
ice-9/eval.scm:223:20: In procedure proc:
error: systoleos-base: unbound variable
hint: Did you forget a `use-modules' form?
```

RafaelPalomar commented:
I guess you could use as many -L flags as needed. Wouldn't this be enough?

oeramo committed:
[BUG][systoleos] Fix kernel panic by not overriding initrd with just microcode

oeramo committed:
Create README.md

oeramo committed:
[ENH][systoleos] Switch guix-systole repo; fix tip in comments

oeramo committed:
[ENH][systoleos] Add Slicer to menu launcher

oeramo committed:
[ENH][systoleos] Autostart Slicer

oeramo committed:
[ENH][systoleos] Automatically set X11 output to preferred

oeramo committed:
[ENH][systoleos] Autolaunch Slicer with a service instead

oeramo committed:
[ENH][systoleos] Full image autostarts Slicer by appending to startup script

oeramo committed:
[ENH][systoleos] Remove imports from Nonguix channel in base image

oeramo committed:
[ENH][systoleos] Remove Nonguix channel definition in base image

oeramo committed:
[BUG][systoleos] Don't use feh to set background image

oeramo committed:
[ENH][systoleos] Remove dependency from Guix mirror, fix branch for guix-systole channel

oeramo committed:
[STYLE][systoleos] Add clarification about Nonguix channel in README

RafaelPalomar commented:
@oeramo, what's the status of this? Is it ready for review?

oeramo committed:
[BUG][systoleos] Fix path to background picture

oeramo committed:
[WIP][systoleos] Remove wrong launch of Slicer

oeramo committed:
[ENH][systoleos] Install Slicer and module as user packages; Fix sudo for user

Commit:
[ENH][deployment] Add deployment configuration file for SystoleOS

Commit:
[STYLE][deployment] Improve comments for ssh target

Commit:
[ENH][deployment] Adapt file system to a system

Commit:
[STYLE][deployment] Add explaining comments regarding adapting deployment file

asiangoldfish committed:
[WIP][systoleos] Remove conky and dicomlink

asiangoldfish committed:
[WIP][deployment] Include SSH and fix GRUB mounting point

asiangoldfish committed:
[BUG] Remove channels file from deployment definitions

asiangoldfish commented:
Hi @RafaelPalomar. The PR is ready for review.

As for refactoring the code so there is one base class and one can inherit from it for special purposes, I have opened a new issue at #2 to address this case.
